[ADD] events: revenues report (#9953)
odoo/documentation: content/applications/marketing/events/revenues_report.rst

===============
Revenues report
===============

> The Odoo *Events* application provides the ability to create custom reports, based on event-related

Reviewer: ```suggestion
The Odoo *Events* application allows users to create custom reports, based on event-related
```

Author: I've been instructed from ZST to avoid using the term "users" whenever possible. Hence, my wording here.
![Screenshot 2024-05-30 1221011](https://github.com/odoo/documentation/assets/73958186/effa68a3-1331-4f09-b09a-183a7d4ff2ca)

data and analytics. These reports can either be focused on the *Attendees* or the *Revenues*. Both
reporting pages offer their own unique array of metrics and options.

The following documentation focuses solely on the reporting options related to event *Revenues*.

Revenues reporting page
=======================

To access the *Attendees* reporting page in the Odoo *Events* application, navigate to
:menuselection:`Events app --> Reporting --> Revenues`.

.. image:: revenues_report/default-view.png
   :align: center
   :alt: The default view of the Revenues reporting page in the Odoo Events application.

By default, the :guilabel:`Revenues` reporting page appears as a graph (a :icon:`fa-line-chart`
:guilabel:`(line chart)` with :icon:`fa-database` :guilabel:`(stacked)` data). The default filters,
:guilabel:`Non-free tickets` and :guilabel:`Event Start Date: (current year)`, are present in the
:guilabel:`Search...` bar.

.. tip::
   To learn more about the various graph views (and graph view options), refer to the documentation
   on :ref:`Graph views <reporting/using-graph>`.

> The :guilabel:`Attendees` reporting page can also be viewed as a :ref:`pivot table
> <reporting/views/pivot>`, by clicking the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon in
> the upper-right corner.

Reviewer: ```suggestion
The :guilabel:`Revenues` reporting page can also be viewed as a :ref:`pivot table
<reporting/views/pivot>`, by clicking the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon in
the upper-right corner.
```
I would move this above the tip block.

Author: While I fully appreciate this suggestion, I feel like, since this portion is discussing the pivot table view...it seems a bit "out of place" to put it above a graph-related admonition. So, taking that into consideration, I think I am going to leave this placement as-is...but, if I get a similar note down the line, I'll be sure to implement the suggestion (and give YOU 100% credit for suggesting it first! 👍 )

Measures
--------

Choosing specific :ref:`Measures <reporting/choosing-measures>` is a quick way to customize
reporting pages.

Regardless of view (graph or pivot table), the measures on the :guilabel:`Revenues` reporting page
are as follows: :guilabel:`Revenues`, :guilabel:`Untaxed Revenues`, and :guilabel:`Count`.

.. note::
   In the default graph view of the :guilabel:`Revenues` reporting page, only the
   :guilabel:`Revenues` option is set in the :guilabel:`Measures` drop-down menu.

   When the pivot option is selected, via the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon
   in the upper-right corner, all :guilabel:`Measures` options are selected, by default.

.. tip::

>    In graph view, only one of the :guilabel:`Measures` can be selected at a time.

Reviewer: I think these two blocks can be combined...
In the default graph view, only one of the measures can be selected at a time, the revenues option is selected by default, etc.
In the pivot view, etc...

- :guilabel:`Revenues`: shows the revenues generated from events.
- :guilabel:`Untaxed Revenues`: shows the untaxed revenues generated from events.
- :guilabel:`Count`: shows the total amount of registrants who attended events.

Filters and grouping options
----------------------------

> To reveal a drop-down menu of filter and grouping options that can be utilized to create custom
> revenue reports, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the
> :guilabel:`Search...` bar.
> 
> Doing so opens a drop-down mega menu of options organized into columns: :ref:`Filters
> <search/preconfigured-filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites
> <search/favorites>`.

Reviewer: ```suggestion
Click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the
:guilabel:`Search...` bar. Doing so opens a drop-down mega menu of options organized into columns: :ref:`Filters
<search/preconfigured-filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites
<search/favorites>`.
opens a drop-down mega menu of options organized into columns: :ref:`Filters
<search/preconfigured-filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites
<search/favorites>`.
``````

Author: Personally, I prefer to provide an abbreviated context to let readers know *what* they can do before telling them *how* to do it. But, same as above, if I get the same note in a different stage, I'll be sure to implement it (w/ you getting all the credit for the original suggestion 😄 )

.. note::

>    If time-related option has been selected from the :guilabel:`Filters` column (e.g. the default

Reviewer: I don't know if you want to include this, but only one selection can be made from the comparison column at a time. (eg- you can compare start dates from this year vs last year, but you can't compare start dates and registration dates)

  events.
- :guilabel:`Add Custom Filter`: create a custom filter to analyze event-related revenue data. To
  learn more, refer to the documentation on :ref:`custom filters <search/custom-filters>`.

Group By options
~~~~~~~~~~~~~~~~

In the :guilabel:`Group By` column of the drop-down mega menu, there are various event-related
options that can be utilized to create custom groupings of data for reports.

Multiple :guilabel:`Group By` options can be selected at once.

The :guilabel:`Group By` column has the following options:

- :guilabel:`Event Type`: group data based on the type of event.
- :guilabel:`Event`: organize data into individual groups, separated by events.
- :guilabel:`Product`: group data based on the event registration product.
- :guilabel:`Ticket`: group data based on the type of event ticket purchased by attendees.
- :guilabel:`Registration Status`: group data based on the status of registrations.
- :guilabel:`Sale Order Status`: group data based on the status of event-related sales orders.
- :guilabel:`Customer`: group data based on customer records.
- :guilabel:`Add Custom Group`: click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to

>   reveal a drop-down of grouping options to choose from. To select one, simply click on the desired

Reviewer: ```suggestion
  reveal a drop-down of grouping options to choose from. To select one, click on the desired
```

====================================

The following is an example of how various filters and grouping options can be utilized to create a
useful analytic report related to event revenues. In this case, the configurations present data
about sold or free tickets to published events, with the metrics separated by ticket type and event.

.. image:: revenues_report/event-ticket-analysis.png
   :align: center
   :alt: The event ticket analysis sample report with unique filters and groupings in place.

To create such a report, stay in the default graph view, but remove the default filters from the
:guilabel:`Search...` bar.

Then, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the
:guilabel:`Search...` bar, to reveal the drop-down mega menu of filter and grouping options.

From here, select the :guilabel:`Free` and :guilabel:`Sold` from the :guilabel:`Filters` column.

Then, since it is desired to **only** view data related to already published events, select the
:guilabel:`Published Events` option in the :guilabel:`Filters` column, as well.

Next, in the :guilabel:`Group By` column, select the :guilabel:`Event` and :guilabel:`Ticket`

> options, in that sequential order. Doing so first groups the data by event *then* by ticket type,
> and not the other way around, which provides a more useful array of data to analyze.

Reviewer: ```suggestion
options, in that sequential order. Doing so first groups the data by event *then* by ticket type, which provides a more useful array of data to analyze.
```
It may be worth a tip block about how the order of groups are selected impacts the results. (I realize this falls under more of the "essentials" docs, but I don't think it is a problem to add a quick mention of it).

From there, additional configurations can be added for more detailed data, if desired.

If no additional filters or groupings are added, Odoo presents a graphical representation of data
related to all *free* or *sold* tickets for *published events*, grouped by *event*, and organized by

> *ticket* type.

Reviewer: While I think the graph is a great visual, if possible it would be nice to also see the pivot view for this report. (I know the reports can be hard to mock so if it is too much to fake it, don't worry about it, but I think it would be interesting to see both possibilities).

- :guilabel:`Untaxed Revenues`: shows the untaxed revenues generated from events.
- :guilabel:`Count`: shows the total amount of registrants who attended events.

Filters and grouping options
----------------------------

To reveal a drop-down menu of filter and grouping options to create custom reports, click the
:icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the search bar.

Doing so opens a drop-down mega menu of options organized into columns: :ref:`Filters
<search/preconfigured-filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites
<search/favorites>`.

.. note::
   If a time-related option has been selected from the :guilabel:`Filters` column (e.g. the default
   :guilabel:`Event Start Date: (year)` filter), a :guilabel:`Comparison` column appears, with
   comparison options for the corresponding time-related filter option selected.

   Only **one** selection can be made from the :guilabel:`Comparison` column at a time.

.. seealso::
   :doc:`../../essentials/search`

Filter options
~~~~~~~~~~~~~~

In the :guilabel:`Filters` column of the drop-down mega menu, there are various event-related

> options that can be utlizied to create custom reports, based on a number of specific criteria.

Reviewer 2: Small typo fix :)
```suggestion
options that can be utilized to create custom reports, based on a number of specific criteria.
```

free tickets to published events, with the metrics separated by ticket type and event.

.. image:: revenues_report/event-ticket-analysis.png
   :align: center
   :alt: The event ticket analysis sample report with unique filters and groupings in place.

To create such a report, navigate to :menuselection:`Events app --> Reporting --> Revenues`. Stay in
the default graph view, but remove the default filters from the search bar.

Then, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the search bar, to
reveal the drop-down mega menu of filter and grouping options.

From here, select :guilabel:`Free` and :guilabel:`Sold` from the :guilabel:`Filters` column.

Then, since it is desired to **only** view data related to already published events, select the
:guilabel:`Published Events` option in the :guilabel:`Filters` column, as well.

Next, in the :guilabel:`Group By` column, select the :guilabel:`Event` and :guilabel:`Ticket`
options, **in that sequential order**. Doing so first groups the data by event, *then* by ticket
type, which provides a more useful array of data to analyze.

> .. important::
>   The order in which the options are selected in the :guilabel:`Group By` column directly affects
>   how the data is presented on the report.

Reviewer 2: Add one more space for admonition alignment :)
```suggestion
.. important::
   The order in which the options are selected in the :guilabel:`Group By` column directly affects
   how the data is presented on the report.
```

From there, additional configurations can be added for more detailed data, if desired.

If no additional filters or groupings are added, Odoo presents a graphical representation of data
related to all *free* or *sold* tickets for *published events*, grouped by *event*, and organized by
*ticket* type.

Sample report: event type analysis (pivot table)
================================================

The following is an example of how various filters and grouping options can create a useful analytic
pivot table report related to event revenues. In this case, the configurations present data about
how much revenue different event types have generated, in order to gauge which events are the most
profitable.

.. image:: revenues_report/event-type-analysis.png
   :align: center
   :alt: The event type analysis sample report with unique filters and groupings in place.

First, navigate to :menuselection:`Events app --> Reporting --> Revenues`, and switch to the pivot
table view, by clicking the :icon:`oi-view-pivot` :guilabel:`(Pivot)` icon in the upper-right
corner.

> Keep the default fitlers (:guilabel:`Non-free tickets` and :guilabel:`Event Start Date: (year)`) in

Reviewer 2: One more quick fix 
```suggestion
Keep the default filters (:guilabel:`Non-free tickets` and :guilabel:`Event Start Date: (year)`) in
```

===============
Revenues report
===============

The Odoo *Events* application provides the ability to create custom reports, based on event-related
data and analytics. These reports can either be focused on the *Attendees* or the *Revenues*. Both
reporting pages offer their own unique array of metrics and options.

The following documentation focuses solely on the reporting options related to event *Revenues*.

Revenues reporting page
=======================

To access the *Attendees* reporting page in the Odoo *Events* application, navigate to
:menuselection:`Events app --> Reporting --> Revenues`.

.. image:: revenues_report/default-view.png
   :align: center
   :alt: The default view of the Revenues reporting page in the Odoo Events application.

By default, the :guilabel:`Revenues` reporting page appears as a graph (a :icon:`fa-line-chart`

> :guilabel:`(line chart)` with :icon:`fa-database` :guilabel:`(stacked)` data). The default filters,

Reviewer 3: ```suggestion
:guilabel:`(Line Chart)` with :icon:`fa-database` :guilabel:`(Stacked)` data). The default filters,
```
they rollovers are in title case- totally optional, just suggesting.

:guilabel:`Non-free tickets` and :guilabel:`Event Start Date: (current year)`, are present in the
:guilabel:`Search...` bar.

.. tip::
   To learn more about the various graph views (and graph view options), refer to the documentation
   on :ref:`Graph views <reporting/using-graph>`.

The :guilabel:`Revenues` reporting page can also be viewed as a :ref:`pivot table

> <reporting/views/pivot>`, by clicking the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon in

Reviewer 3: ```suggestion
<reporting/views/pivot>`, by clicking the :icon:`oi-view-pivot` :guilabel:`(Pivot)` icon in
```
The rollover only displays Pivot- but either is fine

the upper-right corner.

Measures
--------

Choosing specific :ref:`Measures <reporting/choosing-measures>` is a quick way to customize
reporting pages.

Regardless of view (graph or pivot table), the measures on the :guilabel:`Revenues` reporting page
are as follows: :guilabel:`Revenues`, :guilabel:`Untaxed Revenues`, and :guilabel:`Count`.

.. note::
   In the default graph view of the :guilabel:`Revenues` reporting page, only the
   :guilabel:`Revenues` option is set in the :guilabel:`Measures` drop-down menu.

   In graph view, only one of the :guilabel:`Measures` can be selected at a time.

>    When the pivot option is selected, via the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon

Reviewer 3: ```suggestion
   When the pivot option is selected, via the :icon:`oi-view-pivot` :guilabel:`(Pivot)` icon
```

   in the upper-right corner, all :guilabel:`Measures` options are selected, by default.

- :guilabel:`Revenues`: shows the revenues generated from events.
- :guilabel:`Untaxed Revenues`: shows the untaxed revenues generated from events.

> - :guilabel:`Count`: shows the total amount of registrants who attended events.

Reviewer 3: Question- if events are scheduled in the future and tickets are sold but the event has not happened yet- do they also appear on this graph by default?

Author: It is my understanding that, since this is **purely** a Revenues reporting page, it will provide data on all revenue generated from events (past, present, or future). 👍

Filters and grouping options
----------------------------

To reveal a drop-down menu of filter and grouping options that can be utilized to create custom
revenue reports, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the
:guilabel:`Search...` bar.

> Doing so opens a drop-down mega menu of options organized into columns: :ref:`Filters

Reviewer 3: You could add the icons if you want before the words? Totally optional! I just love icons =D

<search/preconfigured-filters>`, :ref:`Group By <search/group>`, and :ref:`Favorites
<search/favorites>`.

.. note::

>    If time-related option has been selected from the :guilabel:`Filters` column (e.g. the default

Reviewer 3: ```suggestion
   If a time-related option has been selected from the :guilabel:`Filters` column (e.g. the default
```

   :guilabel:`Event Start Date: (year)` filter), a :guilabel:`Comparison` column appears, with

>    unique comparison options for the corresponding time-related filter option(s) selected.

Reviewer 3: ```suggestion
   unique comparison options for the corresponding time-related filter options selected.
```

   Only **one** selection can be made from the :guilabel:`Comparison` column at a time.

.. seealso::
   :doc:`../../essentials/search`

Filter options
~~~~~~~~~~~~~~

In the :guilabel:`Filters` column of the drop-down mega menu, there are various event-related
options that can be utlizied to create custom reports, based on a number of specific criteria.

Multiple options in the :guilabel:`Filters` column can be selected at once.

The :guilabel:`Filters` column has the following options:

- :guilabel:`Non-free tickets`: event tickets/registrations that were **not** free.
- :guilabel:`Free`: event tickets/registrations that *were* free.

> - :guilabel:`Pending Payment`: event tickets/registrations that were purchased, but still have

Reviewer 3: ```suggestion
- :guilabel:`Pending payment`: event tickets/registrations that were purchased, but still have
```
that's how it appears- but you can keep it caps if you like

  payment pending.
- :guilabel:`Sold`: event tickets/registrations that have been successfully sold (and paid for).
- :guilabel:`Registration Date`: click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to
  reveal a list of month, quarter, and year options. Select any number of these options to view a

>   specific period(s) of time and see how many registrations happened during that time.

Reviewer 3: ```suggestion
  specific period of time and see how many registrations happened during that time.
```

- :guilabel:`Upcoming/Running`: include revenue-related information for events that are either
  currently running or are going to happen in the future.
- :guilabel:`Past Events`: include revenue-related information for events that have already taken
  place.
- :guilabel:`Event Start Date`: click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to
  reveal a list of month, quarter, and year options. Select any number of these options to designate
  event start dates to use as filters for revenue-related event data.
- :guilabel:`Event End Date`: click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon to
  reveal a list of month, quarter, and year options. Select any number of these options to designate
  event end dates to use as filters for revenue-related event data.
- :guilabel:`Published Events`: select this option to show revenue-related data for published
  events.
- :guilabel:`Add Custom Filter`: create a custom filter to analyze event-related revenue data. To
  learn more, refer to the documentation on :ref:`custom filters <search/custom-filters>`.

Group By options
~~~~~~~~~~~~~~~~

> In the :guilabel:`Group By` column of the drop-down mega menu, there are various event-related

Reviewer 3: ```suggestion
In the :guilabel:`Group By` column of the :guilabel:`Search...` drop-down mega menu, there are various event-related
```

  reveal a drop-down of grouping options to choose from. To select one, click on the desired option
  from the drop-down menu, and Odoo adds it to the :guilabel:`Group By` column. Multiple selections
  can be made.

Sample report: event ticket analysis (graph)
============================================

The following is an example of how various filters and grouping options can be utilized to create a
useful analytic graph report related to event revenues. In this case, the configurations present
data about sold or free tickets to published events, with the metrics separated by ticket type and
event.

.. image:: revenues_report/event-ticket-analysis.png
   :align: center
   :alt: The event ticket analysis sample report with unique filters and groupings in place.

To create such a report, navigate to :menuselection:`Events app --> Reporting --> Revenues'. Stay in
the default graph view, but remove the default filters from the :guilabel:`Search...` bar.

Then, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of the
:guilabel:`Search...` bar, to reveal the drop-down mega menu of filter and grouping options.

> From here, select the :guilabel:`Free` and :guilabel:`Sold` from the :guilabel:`Filters` column.

Reviewer 3: ```suggestion
From here, select :guilabel:`Free` and :guilabel:`Sold` from the :guilabel:`Filters` column.
```

Then, since it is desired to **only** view data related to already published events, select the
:guilabel:`Published Events` option in the :guilabel:`Filters` column, as well.

Next, in the :guilabel:`Group By` column, select the :guilabel:`Event` and :guilabel:`Ticket`

> options, in that sequential order. Doing so first groups the data by event *then* by ticket type,

Reviewer 3: ```suggestion
options, **in that sequential order**. Doing so first groups the data by event *then* by ticket type,
```
Just because it seems important!

and not the other way around, which provides a more useful array of data to analyze.

.. important::
  The order in which the options are selected in the :guilabel:`Group By` column directly affects

>   how the data is presented (and visualized) on the report.

Reviewer 3: ```suggestion
  how the data is presented, and visualized, on the report.
```

From there, additional configurations can be added for more detailed data, if desired.

If no additional filters or groupings are added, Odoo presents a graphical representation of data
related to all *free* or *sold* tickets for *published events*, grouped by *event*, and organized by

> *ticket* type.

Reviewer 3: General question - only because I am not the best at reports and the visuals - is it more clear to someone to view this data in a bar chart? Or is it more clear in the graph? Just honestly asking, as I am not sure which is easier to process. For me, I do like the bars. I honestly don't know which is easier/the default for most people who view reports all day =D

Author: Great question, and it really all comes to down to personal preference. It's a true "case by case" basis, in my opinion. And, since this is merely a sample of how users can utilize this reporting page, I believe any workable example should suffice.

Sample report: event type analysis (pivot table)
================================================

The following is an example of how various filters and grouping options can be utilized to create a
useful analytic pivot table report related to event revenues. In this case, the configurations
present data about how much revenue different event types have generated, in order to gauge which

> events are the most profitable.

Reviewer 3: Do you want to perhaps add that in the pivot, they can export the data and add to dashboards or reports within Odoo - and that could be a value add, or why someone would want to showcase the data in a pivot rather than a graph?

Author: I fully understand (and appreciate) the detailed suggestion. I completely get where you're coming from. However, I feel like that sort of addition may "muddy the waters" a bit. At this point in the doc, I've provided the user/reader with adequate information to do as they please on this reporting page, and I'm not entirely sure another lengthy tangent of this nature would be super useful, as they may already be aware of such functionality from an essentials doc. But, again, I really appreciate the suggestion and attention to detail!

.. image:: revenues_report/event-type-analysis.png
   :align: center
   :alt: The event type analysis sample report with unique filters and groupings in place.

First, navigate to :menuselection:`Events app --> Reporting --> Revenues`, and switch to the pivot

> table view, by clicking the :icon:`oi-view-pivot` :guilabel:`(pivot table)` icon in the upper-right

Reviewer 3: ```suggestion
table view, by clicking the :icon:`oi-view-pivot` :guilabel:`(Pivot)` icon in the upper-right
```

corner.

Keep the default fitlers (:guilabel:`Non-free tickets` and :guilabel:`Event Start Date: (year)`) in
the :guilabel:`Search...` bar.

Next, open the :guilabel:`Measures` drop-down menu, and deselect the option for :guilabel:`Count`,
because this report is only going to focus on revenues.

Then, click :icon:`fa-plus-square` :guilabel:`Total` above the column titles, and select
:guilabel:`Event Type` from the resulting drop-down menu.

> With these configurations in place, all the revenues generated from the events (and their
> corresponding registrations) are displayed, organized by the event type (present as expandable
> columns).

Reviewer 3: ```suggestion
With these configurations in place, all the revenues generated from the events and their
corresponding registrations are displayed, and organized by the event type, presented as expandable
columns.
```
